### PR TITLE
reword existing addon maintenance document and add reasoning for it

### PR DESCRIPTION
**Pull request by jelhan**
This adds the reasons *why* we have shared conventions for the addons in the org. It also starts using the established *must*, *should* and *may* terminology.

I tried to change as less as possible from the existing recommendations so that the discussion of this PR could focus on terminology and structure. But I needed to do so at some places to clarify what the existing content means.

The pull request is build on top of #7 and #9. Both PRs should be merged before.

This is part of #5.

### CODE REVIEW

File: README.md
Comment (jelhan) [inline README.md:52]: Adding a changelog wasn't a requirement nor a clear recommendation before. But I think at least a recommendation fits better than a *may* for this case. I explicitly added that it may not cover historical versions to reduce the initial work for adding a changelog.

File: README.md
Comment (jelhan) [inline README.md:48]: The JSON:API specification uses quotes to differentiate additional examples and explanations from main specification. I'm using a similar pattern here.

File: README.md
Comment (jelhan) [inline README.md:60]: Before it wasn't a clear recommendation to use the script. But I think it's helpful to have a reference for the recommended configuration and this script work wells for that purpose also.

I plan to add a chapter about release process which recommends to use `release-it` later.

File: README.md
Comment (jelhan) [inline README.md]: I wasn't clear what *learning team's recommendations* refers to. I only found the chapter in contributing docs of `ember-source` repo. But that refers to a third-party document. The text and link doesn't fit well together anymore. Are there any documented recommendations from the learning team that I missed? Otherwise I suggest that we reword:

```suggestion
GitHub issues and pull request should be labeled. It should follow the [rules for ember-source repository](https://github.com/emberjs/ember.js/blob/master/CONTRIBUTING.md#issue-labeling) and [lerna-changelog](https://github.com/lerna/lerna-changelog)'s defaults.
```

File: README.md
Comment (jelhan) [inline README.md]: This might be the most fundamental change to the meaning. I thought we should explicitly state that the code of conduct apply to all addons under the org regardless if it's explicitly mentioned in the repo. I also thought that it should point to ember community guidelines instead of the generic Contributor Covenant. But to be honest I'm not an expert on the topic and open to change back if this doesn't make much sense.

/cc @efx

Comment (ghost) [reply README.md]: That is a great idea to defer to the community specific expectations; I had not considered that in detail. I'm with you not being an expert on the topic 😄

File: README.md
Comment (ghost) [reply README.md]: This was my ambiguous thought! I was thinking that the learning team as one of the ember core teams spends a lot of time thinking about how to best present topics and manage contributions. That said, it doesn't necessarily follow that they've built a taxonomy for their labels. So I intended to say "whatever ember core project that has a well defined issue labeling convention". I'd hope for us to share work since the OSS dimension to addons seems quite similar to the core projects.

File: README.md
Comment (MelSumner) [inline README.md]:
```suggestion
- After the checklist is complete, transfer the repository to one of them here on GitHub.
```

File: README.md
Comment (MelSumner) [inline README.md]:
```suggestion
- [ ] Add at least one of the adopted-ember-addons org owners (`melsumner`, `alonbukai`, or `knownasilya`) to the maintainers of the package on `npm`. Make sure they have all admin permissions. Remove other maintainers, if reasonable.
```

**Comment by MelSumner** [comment]
@jelhan @Alonski we also need to start enforcing 2FA for anyone who wants to participate in maintaining apps. It's required for other Ember repos and it's a sensible default. But we'll need to reach out to the folks who don't have it enabled yet and give them a heads up.

File: README.md
Comment (jelhan) [inline README.md:66]: It might be confusing to have to different sets of tags for issues and pull requests. We may want to revisit this part later. I clarified which one should be applied to which to avoid confusion. The _StandardIssueLabes_ used by ember-source repository and defaults of lerna-changelog are _not_ compatible.

**Comment by jelhan** [comment]
Rebased after #7 and #9 have been merged. This should be know way easier to review.

**Comment by bendemboski** [comment]
Looks good to me! Thanks largely to @rwwagner90 we're already hitting almost all of the `should`s in `ember-electron` -- might be able to improve our issue labeling though.

**Review by Alonski — APPROVED** [review]
Awesome work @jelhan !!!

**Comment by knownasilya** [comment]
@jelhan is this ready for merge?

**Comment by jelhan** [comment]
> @jelhan is this ready for merge?

Yes.

**Comment by knownasilya** [comment]
Should we have prettier here? Can probably copy it from the prettier rfc by @rwjblue

**Comment by jelhan** [comment]
> Should we have prettier here? Can probably copy it from the prettier rfc by @rwjblue

Already created a draft for prettier in #14. I think it's better to discuss each topic separately after terminology and structure has been settled.

**Pull request merged**
